packages/ses/src/module-proxy.js

```diff
@@ -0,0 +1,173 @@
+          throw new TypeError(
+            `Cannot get property ${q(
+              name,
+            )} of module exports namespace, the module has not yet begun executing`,
+          );
+        }
+        return Reflect.get(internalModule, name, receiver);
+      },
+      set(_target, name, value) {
+        if (internalModule === undefined) {
+          throw new TypeError(
+            `Cannot set property ${q(
+              name,
+            )} of module exports namespace, the module has not yet begun executing`,
+          );
+        }
+        if (!Reflect.set(internalModule, name, value)) {
+          throw new TypeError(
+            `Cannot add property ${q(
+              name,
+            )}, the module exports namespace is not extensible`,
+          );
+        }
+        return true;
+      },
+      has(_target, name) {
+        if (internalModule === undefined) {
+          throw new TypeError(
+            `Cannot check property ${q(
+              name,
+            )}, the module has not yet begun to execute`,
+          );
+        }
+        return Reflect.has(target, name);
+      },
+      deleteProperty(_target, name) {
+        throw new TypeError(
+          `Cannot delete property ${q(
+            name,
+          )}, the module exports namespace is not extensible`,
+        );
+      },
+      ownKeys(_target) {
+        if (internalModule === undefined) {
+          throw new TypeError(
+            `Cannot enumerate keys, the module has not yet begun to execute`,
+          );
+        }
+        return Reflect.ownKeys(internalModule);
+      },
+      getOwnPropertyDescriptor(_target, name) {
+        if (internalModule === undefined) {
+          throw new TypeError(
+            `Cannot get own property descriptor ${q(
+              name,
+            )}, the module has not yet begun to execute`,
+          );
+        }
+        return Reflect.getOwnPropertyDescriptor(internalModule, name);
+      },
+      preventExtensions(_target) {
+        if (internalModule === undefined) {
+          throw new TypeError(
+            `Cannot prevent extensions to module exports namespace until the module has not yet begun to execute`,
+          );
+        }
+        preventExtensions(target);
+        preventExtensions(internalModule);
+        return true;
+      },
+      isExtensible() {
+        if (internalModule === undefined) {
+          return Reflect.isExtensible(target);
+        }
+        return Reflect.isExtensible(internalModule);
+      },
+      getPrototypeOf(_target) {
+        return null;
+      },
+      setPrototypeOf(_target, _proto) {
+        throw new TypeError(`Cannot set prototype of module exports namespace`);
+      },
+      defineProperty(_target, name, _descriptor) {
+        throw new TypeError(
+          `Cannot assign to read-only property ${q(
+            name,
+          )} of module exports namespace, module exports are not configurable`,
```

Comment:
It does make sense for the handler to reject all `defineProperty` calls, so the unconditional error here is appropriate.

However, even for non-configurable properties of non-exotic objects, some `defineProperty` calls are allowed, such as those that change the `value` or writable non-configurable data properties. So this error message can be wrong.

Reply:
Making this more correct by saying less.